terraform-environments/hosts-user-data.sh

```diff
@@ -7,6 +7,9 @@
   curl_path=/bin/curl
   sed_path=/bin/sed
 
+# store script encoded in base64 so the ec2 has access to it
+  encoded_python_script="IyBVc2UgdGhpcyBjb2RlIHNuaXBwZXQgaW4geW91ciBhcHAuCiMgSWYgeW91IG5lZWQgbW9yZSBpbmZvcm1hdGlvbiBhYm91dCBjb25maWd1cmF0aW9ucwojIG9yIGltcGxlbWVudGluZyB0aGUgc2FtcGxlIGNvZGUsIHZpc2l0IHRoZSBBV1MgZG9jczoKIyBodHRwczovL2F3cy5hbWF6b24uY29tL2RldmVsb3Blci9sYW5ndWFnZS9weXRob24vCgppbXBvcnQgYm90bzMKZnJvbSBib3RvY29yZS5leGNlcHRpb25zIGltcG9ydCBDbGllbnRFcnJvcgoKCmRlZiBnZXRfc2VjcmV0KCk6CgogICAgc2VjcmV0X25hbWUgPSAicHJvZC9ob3N0cy9zZWNyZXRzIgogICAgcmVnaW9uX25hbWUgPSAidXMtZWFzdC0xIgoKICAgICMgQ3JlYXRlIGEgU2VjcmV0cyBNYW5hZ2VyIGNsaWVudAogICAgc2Vzc2lvbiA9IGJvdG8zLnNlc3Npb24uU2Vzc2lvbigpCiAgICBjbGllbnQgPSBzZXNzaW9uLmNsaWVudCgKICAgICAgICBzZXJ2aWNlX25hbWU9J3NlY3JldHNtYW5hZ2VyJywKICAgICAgICByZWdpb25fbmFtZT1yZWdpb25fbmFtZQogICAgKQoKICAgIHRyeToKICAgICAgICBnZXRfc2VjcmV0X3ZhbHVlX3Jlc3BvbnNlID0gY2xpZW50LmdldF9zZWNyZXRfdmFsdWUoCiAgICAgICAgICAgIFNlY3JldElkPXNlY3JldF9uYW1lCiAgICAgICAgKQogICAgZXhjZXB0IENsaWVudEVycm9yIGFzIGU6CiAgICAgICAgIyBGb3IgYSBsaXN0IG9mIGV4Y2VwdGlvbnMgdGhyb3duLCBzZWUKICAgICAgICAjIGh0dHBzOi8vZG9jcy5hd3MuYW1hem9uLmNvbS9zZWNyZXRzbWFuYWdlci9sYXRlc3QvYXBpcmVmZXJlbmNlL0FQSV9HZXRTZWNyZXRWYWx1ZS5odG1sCiAgICAgICAgcmFpc2UgZQoKICAgICMgRGVjcnlwdHMgc2VjcmV0IHVzaW5nIHRoZSBhc3NvY2lhdGVkIEtNUyBrZXkuCiAgICBzZWNyZXQgPSBnZXRfc2VjcmV0X3ZhbHVlX3Jlc3BvbnNlWydTZWNyZXRTdHJpbmcnXQoKICAgICMgY3JlYXRlIGEganNvbiBvYmplY3QgdXNpbmYgdGhlIHJldHVybmVkIHNlY3JldAogICAgZGF0YSA9IGpzb24ubG9hZHMoc2VjcmV0KQoKICAgICMgR2V0IHRoZSB2YWx1ZSBvZiBlYWNoIGtleQogICAgcHVibGljX2tleSA9IGRhdGFbJ3B1YmxpYyddCiAgICBwcml2YXRlX2tleSA9IGRhdGFbJ3ByaXZhdGUnXQoKICAgICMgWW91ciBjb2RlIGdvZXMgaGVyZS4KICAgIHByaW50KHB1YmxpY19rZXkpCiAgICBwcmludChwcml2YXRlX2tleSkKCmdldF9zZWNyZXQoKQo"
```

Comment:
Is there a reason to encode this? Just makes it unclear, and can easily get out of sync with the actual script.

Reply:
when researching thats how most of the examples i saw went about it, so i did as well. I think its good like this

Reply:
can you provide a source for why it is necessary? If it works without it we should do without it, because otherwise it is *very* easy for the encoded bit and the normal script to get out of sync - we change the normal script but not the encoded one. Can you try with it decoded?